core/src/main/java/jenkins/model/MasterBuildConfiguration.java

```diff
@@ -51,6 +51,11 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
         Jenkins j = Jenkins.getInstance();
         try {
             // for compatibility reasons, this value is stored in Jenkins
+            String num = json.getString("numExecutors");
+            if (!num.matches("^(\\d+)")) {
+                throw new FormException(Messages.Hudson_Computer_IncorrectNumberOfExecutors(),"numExecutors");
```

Comment:
It would be nicer to just throw the `FormException` from the setter, so you do not need to duplicate code, but I suppose the problem is that this is a checked exception.